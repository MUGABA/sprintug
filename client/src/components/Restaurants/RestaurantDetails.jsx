import { useEffect, useState } from "react";
import { useParams } from "react-router-dom";
import { getDetailsById } from "../../Services/restService";

const RestaurantDetails = () => {
  const [details, setDetails] = useState({});

  const { restId } = useParams();

  const getRestDetails = async () => {
    const { data } = await getDetailsById(restId);

    setDetails(data);
  };

  useEffect(() => {
    getRestDetails();
  }, [restId]);

  const {
    name,
    location,
    image,
    createdAt,
    cuisineType,
    // createdBy: { name: creator },
  } = details;
  return (
    <div>
      <div className="res-details">
        <div className="image-part">
          <h3 data-testid="rest-name-id">Restaurant Details </h3>
          <img src={`${process.env.PUBLIC_URL}/uploads/${image}`} alt="image" />
        </div>

        <div className="details-part">
          <div className="detail">
            <p className="label">name:</p>
            <p className="value">{name}</p>
          </div>

          <div className="detail">
            <p className="label">Cuisine Type:</p>
            <p className="value">{cuisineType}</p>
          </div>
          <div className="detail">
            <p className="label">Location:</p>
            <p className="value">{location}</p>
          </div>
          {/* <div className="detail">
          <p className="label">Created By:</p>
          <p className="value">{creator}</p>
        </div> */}
          <div className="detail">
            <p className="label">Created On:</p>
            <p className="value">{createdAt}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default RestaurantDetails;
