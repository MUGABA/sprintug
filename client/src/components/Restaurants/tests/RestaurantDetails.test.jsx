import { render } from "@testing-library/react";
import React from "react";
import { act } from "react-dom/test-utils";
import RestaurantDetails from "../RestaurantDetails";

RestaurantDetails;

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

it("renders Restaurant Details data", async () => {
  const fakeRestaurant = {
    name: "Joni Baez",
    location: "Kampala Uganda",
    cuisineType: "Chinese Cuisine",
    image: "",
  };

  jest.spyOn(global, "fetch").mockImplementation(() =>
    Promise.resolve({
      json: () => Promise.resolve(fakeRestaurant),
    })
  );

  // Use the asynchronous version of act to apply resolved promises
  await act(async () => {
    render(<RestaurantDetails />, container);
  });

  expect(container).toMatchSnapshot();

  global.fetch.mockRestore();
});
