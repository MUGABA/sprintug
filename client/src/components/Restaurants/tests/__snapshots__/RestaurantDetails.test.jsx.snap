// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Restaurant Details data 1`] = `<div />`;
